server/routerlicious/packages/services-core/src/database.ts

```diff
@@ -140,6 +140,14 @@ export interface ICollection<T> {
     createTTLIndex?(index: any, mongoExpireAfterSeconds?: number): Promise<void>;
 }
 
+export interface IRetryAble {
+    retryEnabled: boolean;
+}
+
+export function isRetryAble<T>(collection: ICollection<T>): collection is ICollection<T> {
+    return "retryEnabled" in collection;
```

Comment:
Sorry if this is obvious, but how does this work? The part I am confused about is that `ICollection` above does not seem to be related to the new `IRetryAble` interface. Also, how does this syntax work? Specifically the `retryEnabled` property from `IRetryAble` being used as a string, and the `in` keyword.

Reply:
I think you come from a C# world too? I was surprised to see `collection is IRetryAble` is not working in typescript too!!!
Here are the details:
https://stackoverflow.com/a/46703380
https://stackoverflow.com/a/14426274
https://bobbyhadz.com/blog/typescript-check-if-object-implements-interface